src/cmap/connect.ts

```diff
@@ -79,14 +79,12 @@ export function connect(options: ConnectionOptions, callback: Callback<Connectio
 }
 
 function checkSupportedServer(hello: Document, options: ConnectionOptions) {
+  const maxWireVersion = Number(hello.maxWireVersion);
+  const minWireVersion = Number(hello.minWireVersion);
   const serverVersionHighEnough =
-    hello &&
-    typeof hello.maxWireVersion?.valueOf() === 'number' &&
-    hello.maxWireVersion >= MIN_SUPPORTED_WIRE_VERSION;
+    hello && !Number.isNaN(maxWireVersion) && hello.maxWireVersion >= MIN_SUPPORTED_WIRE_VERSION;
```

Comment:
```suggestion
    !Number.isNaN(maxWireVersion) && hello.maxWireVersion >= MIN_SUPPORTED_WIRE_VERSION;
```

If hello is nullish which I don't think it can be then the lines 82/83 will throw, so we can remove this check

Reply:
Updated.